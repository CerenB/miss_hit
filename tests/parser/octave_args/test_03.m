% This test is extracted from the GNU Octave testsuite
% For a full copyright please refert to args.tst from GNU Octave

% no inputs, one of multiple outputs
function [x, y] = f ()
  assert (nargin == 0);
  assert (nargout == 1);
  x = 2;
end

function test()
 assert (f () == 2);
end
