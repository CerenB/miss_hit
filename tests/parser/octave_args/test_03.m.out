----------------------------------------------------------------------
--  Parse tree for test_03.m
Root: Function_File
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <f>
      Outputs: Identifier <x>
      Outputs: Identifier <y>
    Body: Sequence_Of_Statements
      Statements: Naked_Expression_Statement
        Expression: Reference
          Name: Identifier <assert>
          Arguments: Binary_Operation ==
            LHS: Identifier <nargin>
            RHS: Number_Literal <0>
      Statements: Naked_Expression_Statement
        Expression: Reference
          Name: Identifier <assert>
          Arguments: Binary_Operation ==
            LHS: Identifier <nargout>
            RHS: Number_Literal <1>
      Statements: Simple_Assignment_Statement
        LHS: Identifier <x>
        RHS: Number_Literal <2>
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <test>
    Body: Sequence_Of_Statements
      Statements: Naked_Expression_Statement
        Expression: Reference
          Name: Identifier <assert>
          Arguments: Binary_Operation ==
            LHS: Reference
              Name: Identifier <f>
            RHS: Number_Literal <2>
----------------------------------------------------------------------
In test_03.m, line 5
| function [x, y] = f ()
|          ^ info: this token is not linked to the ast
In test_03.m, line 5
| function [x, y] = f ()
|               ^ info: this token is not linked to the ast
In test_03.m, line 5
| function [x, y] = f ()
|                 ^ info: this token is not linked to the ast
In test_03.m, line 9
| end
| ^^^ info: this token is not linked to the ast
In test_03.m, line 13
| end
| ^^^ info: this token is not linked to the ast
MISS_HIT Summary: 1 file(s) analysed, everything seemes fine
